modular_skyrat/modules/modular_implants/code/nifsoft_catalog.dm

```diff
@@ -0,0 +1,102 @@
+GLOBAL_LIST_INIT(available_nifsofts, list(
+	/datum/nifsoft/hivemind,
+	/datum/nifsoft/summoner,
+	/datum/nifsoft/shapeshifter,
+))
+
+/datum/computer_file/program/nifsoft_downloader
+	filename = "nifsoftcatalog"
+	filedesc = "NIFSoft Catalog"
+	extended_desc = "A virtual storefront that allows the user to install NIFSofts and purchase various NIF related products"
+	category = PROGRAM_CATEGORY_MISC
+	size = 3
+	tgui_id = "NtosNifsoftCatalog"
+	program_icon = "bag-shopping"
+	usage_flags = PROGRAM_TABLET
+	///What bank account is money being drawn out of?
+	var/datum/bank_account/paying_account
+	///What NIF are the NIFSofts being sent to?
+	var/datum/weakref/target_nif
+
+/datum/computer_file/program/nifsoft_downloader/Destroy(force)
+	paying_account = null
+	target_nif = null
+
+	return ..()
+
+//TGUI STUFF
+
+/datum/computer_file/program/nifsoft_downloader/ui_data(mob/user)
+	var/list/data = list()
+
+	paying_account = computer.computer_id_slot?.registered_account || null
+	data["paying_account"] = paying_account
+
+	return data
+
+/datum/computer_file/program/nifsoft_downloader/ui_static_data(mob/user)
+	var/list/data = get_header_data()
+	var/list/product_list = list()
+
+	var/mob/living/carbon/human/nif_user = user
+	if(nif_user)
+		var/obj/item/organ/internal/cyberimp/brain/nif/user_nif = nif_user.getorgan(/obj/item/organ/internal/cyberimp/brain/nif)
+		if(user_nif)
+			target_nif = user_nif
+		else
+			target_nif = null
```

Comment:
Honestly you could even just not even add the `|| null` here, but I'm putting it there just because I don't know what the getorgan returns.
```suggestion
		target_nif = user_nif || null
```